woocommerce/woocommerce-ios
# [Blaze] Load suggestions & pass info to Edit Ad screen

Part of: #11652 


## Description

This PR adds changes to integrate AI suggestions and injects ad info to the edit ad screen.

#### Changes
- Load AI suggestions from the campaign preview screen
- Use the first suggestion to fill the ad data
- Inject the ad data into the Edit Ad screen
- Update the preview screen with the edited info received from "Edit Ad" screen
- Download the product image asynchronously to pass to the "Edit Ad" screen. 

🗒️  I will handle adding unit tests in a future PR. 

## Testing instructions
- Login into a store with at least one published product
- Open the product detail screen
- Tap on "Promote with Blaze" -> "Create Your Campaign"
- You will land in the Blaze Ad preview screen
- The AI-based suggestions will be loaded and you will see a tagline and description.
- Validate that the first product image is used for the ad.
- Tap on the "Edit ad" button to open the edit screen
- Try editing the ad details. (Image, tagline and description)
- You can also use the "<" and ">" buttons to choose an AI-suggested tagline and description. 
- Tapping "Save" should dismiss the edit screen and update the details in the campaign preview screen

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/1d713ce8-17c3-44e2-9008-d35ae381296b



---
